Error on heavy exchange traffic

arbitrader_1  | 2019-04-09 07:41:56.702 ERROR 1 --- [   scheduling-1] o.s.s.s.TaskUtils$LoggingErrorHandler    : Unexpected error occurred in scheduled task.
arbitrader_1  |
arbitrader_1  | org.knowm.xchange.exceptions.ExchangeException: <h2>This website is under heavy load (queue full)</h2><p>We're sorry, too many people are accessing this website at the same time. We're working on this problem. Please try again later.</p>
arbitrader_1  |  at org.knowm.xchange.quoine.service.QuoineMarketDataServiceRaw.getOrderBook(QuoineMarketDataServiceRaw.java:44)
arbitrader_1  |  at org.knowm.xchange.quoine.service.QuoineMarketDataService.getOrderBook(QuoineMarketDataService.java:67)
arbitrader_1  |  at com.r307.arbitrader.service.TradingService.getLimitPrice(TradingService.java:839)
arbitrader_1  |  at com.r307.arbitrader.service.TradingService.lambda$null$14(TradingService.java:342)
arbitrader_1  |  at java.util.ArrayList.forEach(ArrayList.java:1257)
arbitrader_1  |  at com.r307.arbitrader.service.TradingService.lambda$null$15(TradingService.java:298)
arbitrader_1  |  at java.util.ArrayList.forEach(ArrayList.java:1257)
arbitrader_1  |  at com.r307.arbitrader.service.TradingService.lambda$tick$16(TradingService.java:290)
arbitrader_1  |  at java.util.ArrayList.forEach(ArrayList.java:1257)
arbitrader_1  |  at com.r307.arbitrader.service.TradingService.tick(TradingService.java:290)
arbitrader_1  |  at sun.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
arbitrader_1  |  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
arbitrader_1  |  at java.lang.reflect.Method.invoke(Method.java:498)
arbitrader_1  |  at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84)
arbitrader_1  |  at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
arbitrader_1  |  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
arbitrader_1  |  at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
arbitrader_1  |  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
arbitrader_1  |  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
arbitrader_1  |  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
arbitrader_1  |  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
arbitrader_1  |  at java.lang.Thread.run(Thread.java:748)
arbitrader_1  | Caused by: si.mazi.rescu.HttpStatusIOException: HTTP status code was not OK: 503
arbitrader_1  |  at si.mazi.rescu.ResponseReader.read(ResponseReader.java:104)
arbitrader_1  |  at si.mazi.rescu.RestInvocationHandler.mapInvocationResult(RestInvocationHandler.java:175)
arbitrader_1  |  at si.mazi.rescu.RestInvocationHandler.receiveAndMap(RestInvocationHandler.java:163)
arbitrader_1  |  at si.mazi.rescu.RestInvocationHandler.invoke(RestInvocationHandler.java:119)
arbitrader_1  |  at com.sun.proxy.$Proxy47.getOrderBook(Unknown Source)
arbitrader_1  |  at org.knowm.xchange.quoine.service.QuoineMarketDataServiceRaw.getOrderBook(QuoineMarketDataServiceRaw.java:42)
arbitrader_1  |  ... 21 common frames omitted
arbitrader_1  |



Maybe it will continue after this.... seems like software is still working

--- a/src/main/java/com/r307/arbitrader/service/TradingService.java
+++ b/src/main/java/com/r307/arbitrader/service/TradingService.java
@@ -272,8 +272,14 @@ public class TradingService {
 
         exchanges
             .parallelStream()
-            .forEach(exchange -> getTickers(exchange, getExchangeMetadata(exchange).getTradingPairs())
-                .forEach(ticker -> allTickers.put(tickerKey(exchange, ticker.getCurrencyPair()), ticker)));
+            .forEach(exchange -> {
+                try {
+                    getTickers(exchange, getExchangeMetadata(exchange).getTradingPairs())
+                        .forEach(ticker -> allTickers.put(tickerKey(exchange, ticker.getCurrencyPair()), ticker));
+                } catch (ExchangeException e) {
+                    LOGGER.warn("Failed to fetch ticker for {}", exchange.getExchangeSpecification().getExchangeName());
+                }
+            });
 
         long exchangePollStartTime = System.currentTimeMillis();
 
@@ -327,8 +333,19 @@ public class TradingService {
                     if (maxExposure != null) {
                         BigDecimal longVolume = maxExposure.divide(longTicker.getAsk(), BTC_SCALE, RoundingMode.HALF_EVEN);
                         BigDecimal shortVolume = maxExposure.divide(shortTicker.getBid(), BTC_SCALE, RoundingMode.HALF_EVEN);
-                        BigDecimal longLimitPrice = getLimitPrice(longExchange, currencyPair, longVolume, Order.OrderType.ASK);
-                        BigDecimal shortLimitPrice = getLimitPrice(shortExchange, currencyPair, shortVolume, Order.OrderType.BID);
+                        BigDecimal longLimitPrice;
+                        BigDecimal shortLimitPrice;
+
+                        try {
+                            longLimitPrice = getLimitPrice(longExchange, currencyPair, longVolume, Order.OrderType.ASK);
+                            shortLimitPrice = getLimitPrice(shortExchange, currencyPair, shortVolume, Order.OrderType.BID);
+                        } catch (ExchangeException e) {
+                            LOGGER.warn("Failed to fetch order books for {}/{} to compute entry prices: {}",
+                                longExchange.getExchangeSpecification().getExchangeName(),
+                                shortExchange.getDefaultExchangeSpecification().getExchangeName(),
+                                e.getMessage());
+                            return;
+                        }
 
                         BigDecimal spreadVerification = computeSpread(longLimitPrice, shortLimitPrice);
 

